ts/UpdateAppModal.tsx

```diff
@@ -0,0 +1,137 @@
+/**
+ * A screen where invite the user to update app because the current version is unsupported
+ *
+ */
+
+import { Button, Container, H2, Text, View } from "native-base";
+import * as React from "react";
+import {
+  BackHandler,
+  Image,
+  Linking,
+  Modal,
+  Platform,
+  StyleSheet
+} from "react-native";
+import { connect } from "react-redux";
+import BaseScreenComponent from "./components/screens/BaseScreenComponent";
+import FooterWithButtons from "./components/ui/FooterWithButtons";
+import I18n from "./i18n";
+import customVariables from "./theme/variables";
+
+const styles = StyleSheet.create({
+  text: {
+    marginTop: customVariables.contentPadding,
+    fontSize: 18
+  },
+  container: {
+    margin: customVariables.contentPadding,
+    flex: 1,
+    alignItems: "flex-start"
+  },
+  img: {
+    marginTop: customVariables.contentPaddingLarge,
+    alignSelf: "center"
+  }
+});
+
+class UpdateAppModal extends React.PureComponent<{}> {
+  // No Event on back button android
+  private handleBackPress = () => {
+    return true;
+  };
+
+  public componentDidMount() {
+    BackHandler.addEventListener("hardwareBackPress", this.handleBackPress);
+  }
+
+  public componentWillUnmount() {
+    BackHandler.removeEventListener("hardwareBackPress", this.handleBackPress);
+  }
+
+  private openAppStore() {
```

Comment:
you don't need **_openAppStore_** has to be a class method.
Create a const function outside the class. This could be a possibile implementation:

```typescript
const openAppStore = () => {
  const storeUrl = Platform.select({
    ios: "itms-apps://itunes.apple.com/it/app/testflight/id899247664?mt=8",
    android: "market://details?id=it.teamdigitale.app.italiaapp"
  });
  Linking.openURL(storeUrl).catch(() => {
    // display an toast with an error message
  });
};
```